### PR TITLE
fix(native-filters): Fix native filters config modal

### PR DESCRIPTION
**Pull request by simcha90**
### SUMMARY

This PR fix next flows:
1. There was regression, when you mark filter as `Required` it doesn't shows errors under filter and also not marks it with red (also FilterBar also FiltersConfigModal)

2. When user switch between filter types with Default Value some times it crashes dependent on what filter were changed

3. For filter without `Column` field it doesn't load default value

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/DefaultValue.tsx
Comment (amitmiran137) [inline superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/DefaultValue.tsx:63]: shouldn't this be opposite like 
```suggestion
    !formFilter?.controlValues?.enableEmptyFilter;
```

**Review by amitmiran137 — COMMENTED** [review]
look good
just a small question before merging

File: superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/DefaultValue.tsx
Comment (simcha90) [reply superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/DefaultValue.tsx:63]: it looks strange but it should be like this :)

**Comment by amitmiran137** [comment]
@simcha90 is that behavior valid?
image
on range filter with default value with the full range it doesn't considered to be a default value

**Comment by simcha90** [comment]
> @simcha90 is that behavior valid?
> image
> on range filter with default value with the full range it doesn't considered to be a default value

yes it's valid as discussed with @villebro it means as unset value, it also behavior when you clicks `Clear all`

**Review by amitmiran137 — APPROVED** [review]
LGTM!

tested out all filter types with required and column scenarios and same flow is always the same on every type
 * if no column chosen no default value to choose 
 * if required and default value also checked
 * if required but column then only if save validation message of both column and default value appears like : 
 _Fill all required fields to enable "Default Value"
Default value is required_

**Comment by michael-s-molina** [comment]
The removal of "Value is required" was not a regression. It was a design decision to eliminate the unnecessary spacing among filters. That's why we added the asterisk. This PR reintroduced the text and the spacing.

Screen Shot 2021-07-06 at 7 51 51 AM

@simcha90 Can you please roll back to the previous behavior?

@villebro @junlincc @amitmiran137

**Comment by junlincc** [comment]
@amitmiran137 @simcha90 Hi friends! according to the thread, we are seeing some regressions from this PR. To prevent this from happening Please do leave enough time for community reviews, and we would do the same :). One thing that could help the project is all parties to sync what feature goals we each wanna achieve on a weekly basis. 

We try to stabilize this native filter for prime time, and are investing more resources into QA, we would love to help keeping the PR standard high by testing more, please simply leave us 48 for code reviews and testing.
cc @rosemarie-chiu @graceguo-supercat

**Comment by michael-s-molina** [comment]
Another regression found: the default value Select is not checked when there's a default value.

https://user-images.githubusercontent.com/70410625/124639865-09a6ff80-de63-11eb-9338-2a4e8f613f51.mp4

**Comment by junlincc** [comment]
@simcha90 let's make sure we have both behavior reverted. many thanks 🙏

**Comment by simcha90** [comment]
Hi @junlincc @michael-s-molina yep, you are right it's good point to wait for some more feedbacks before merge, I will take it in account for next PRs

About notes, because this PR fix some other behavioral bugs, I'd prefer create one new PR to fix your notes rather then revert this one, to fix your notes I have some questions:

1. About Spacing / Coloring:
We have here 2 features in one:
a. Adding red colors to filters dependent on the type of filter: like border for Select Filter or make Range Filter is red for required state - this feature is not affecting layout, only change color of the filters
b. Adding `Value is required` text and spacing between items - this feature affecting layout and add spacing, so we have 2 ways here:
- Revert this text for all filters as you proposed in first
- Add this text only when needed and in other cases just remove this space

2. Bug with initial value I'll create separate PR for it

@junlincc @michael-s-molina can you please give your input for first item? And I will create PR ASAP to fix it, thanks

**Comment by michael-s-molina** [comment]
1- About Spacing / Coloring:

a. Keep as it is with the red colors.

b. Remove the text only in the filter bar, not in the modal. The asterisk and the color are sufficient. We don't want the flick effect that happens when we add the text dynamically. 

Thank you.

**Comment by simcha90** [comment]
@michael-s-molina @junlincc 
Hi, I created PR according your proposals: https://github.com/apache/superset/pull/15572
Can you please review it and give your input? Thanks...

**Comment by piewaer** [comment]
Hello, I started front end engineering in the superset-frontend folder, but get lots of errors.
image

1. In the superset-frontend folder, npm ci
2. npm run dev-server
3. 98% of the process, get lots of errors, includes the type error of the data and the error not found by the module

It should be unable to parse JSX, But how do I need to solve it?Looking forward to your reply

**Pull request merged**
